colossalai/shardformer/modeling/llama.py

```diff
@@ -267,11 +273,18 @@ def llama_for_causal_lm_forward(
                 shift_labels = labels[..., 1:].contiguous()
                 # Flatten the tokens
                 loss_fct = CrossEntropyLoss()
-                shift_logits = shift_logits.view(-1, self.config.vocab_size)
                 shift_labels = shift_labels.view(-1)
                 # Enable model parallelism
                 shift_labels = shift_labels.to(shift_logits.device)
-                loss = loss_fct(shift_logits, shift_labels)
+                if shard_config.enable_tensor_parallelism:
+                    tp_world_size = dist.get_world_size(shard_config.tensor_parallel_process_group)
+                    new_vocab_size = self.config.vocab_size // tp_world_size
```

Comment:
No need to compute again. The size of last dim of `shift_logits` is new vocab size